src/common/BaseComponent.ts

```diff
@@ -12,7 +12,7 @@ export class BaseComponent<P, S> extends React.Component<P, S> {
 
   private __async: Async;
   private __events: EventGroup;
-  private __disposables: IDisposable[];
+  private __disposables: (IDisposable | (() => void))[];
```

Comment:
I strongly believe that disposables should always be objects, not bare `dispose` methods.